### PR TITLE
Flesh out the docs

### PR DESCRIPTION
**Pull request by ltratt**
This updates the docs to reflect the recent changes to the actions API, and also adds a lot more documentation to the error recovery section.

### CODE REVIEW

File: doc/src/errorrecovery.md [hunk 1]
```diff
@@ -1,8 +1,396 @@
...
+
+In this case, the first repair sequence caused the input to be parsed as if it
+was equivalent to `2 + 3 * 5`, hence the evaluation of the input to `17`.
+
+
+## Syntax errors and language semantics
+
+Our example inputs so far have deliberately exploited cases where the first
+repair sequence at worst inserted “unimportant” lexemes such as `+` and `*`.
+Since the grammar’s actions never read the contents of such lexemes, only their
```
Comment (ptersilie) [inline doc/src/errorrecovery.md]: I'm wondering if "value" is a better word to describe a lexeme's contents.

File: doc/src/errorrecovery.md [hunk 2]
```diff
@@ -1,8 +1,396 @@
...
+()>`: if we receive `Ok(u64)` we successfully evaluated the expression, but if
+we received `Err(())` we were not able to evaluate the expression. If we
+encounter an integer lexeme which is the result of error recovery, then the
+`INT` lexeme in the second `Factor` action will be `Err(<lexeme>)`. By writing
+`$1.map_err(|_| ())?` we’re saying “if the integer lexeme was created by error
+recovery, percolate `Err(())` upwards”. We then have to tweak a couple of other
+actions to percolate errors upwards, but this is a trivial change.
+
+We then need to make a small tweak to our `main.rs` changing:
+
```
Comment (ptersilie) [inline doc/src/errorrecovery.md]: You could present this as a diff, e.g.:
```diff,ignore
match res {
-    Some(r) => println!("Result: {}", r),
+    Some(Ok(r)) => println!("Result: {}", r),
    _ => eprintln!("Unable to evaluate expression.")
}
```

File: doc/src/errorrecovery.md [hunk 3]
```diff
@@ -28,24 +414,29 @@ Error at line 5 col 21. Repair sequences found:
...
-continue parsing.
+## Turning off error recovery
 
-Error recovery requires no extra help from the user and is turned on
-automatically ([the underlying algorithm is described
-elsewhere](https://arxiv.org/abs/1804.07133)). By default, `Insert` repairs use
-lexeme names from the grammar to tell the user what happened. Optionally users
-can provide simple annotations to provide more intuitive names. `%epp <lexeme
-name> "<pretty print name>"` declarations can be added to the `.y` file e.g.:
+By default, `lrpar` uses the `MF` error recovery algorithm. You can use the
```
Comment (ptersilie) [inline doc/src/errorrecovery.md]: Added commas to improve readbility
```
You can use the `None` error recovery algorithm, which causes parsing to stop as soon as it hits
the first parsing error, with the `recoverer` method in `CTParserBuilder` or
`RTParserBuilder`.
```

**Comment by ptersilie** [comment]
Just a few minor comments.

File: doc/src/errorrecovery.md (hunk 1 above)
Comment (ltratt) [reply doc/src/errorrecovery.md]: Agreed.

File: doc/src/errorrecovery.md (hunk 2 above)
Comment (ltratt) [reply doc/src/errorrecovery.md]: As discussed in person, mdbook doesn't seem to colour diffs in the way we would like.

File: doc/src/errorrecovery.md (hunk 3 above)
Comment (ltratt) [reply doc/src/errorrecovery.md]: Agreed.

**Comment by ltratt** [comment]
See if those comments help.

**Comment by ltratt** [comment]
Squashed (and hopefully fixed!).

**Pull request merged**